checkov/sast/checks_infra/check_parser/base_parser.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC, abstractmethod
+from typing import Dict, Any
+
+from checkov.sast.consts import SemgrepAttribute, CHECKOV_SEVERITY_TO_SEMGREP_SEVERITY
+
+
+class BaseSastCheckParser(ABC):
+    def parse_raw_check_to_semgrep(self, raw_check: Dict[str, Dict[str, Any]], check_file: str | None = None) -> Dict[str, Any]:
+        semgrep_rule: Dict[str, Any] = {}
+        if not self._raw_check_is_valid(raw_check):
+            logging.error(f'cant parse the following policy: {raw_check}')
+        try:
+            semgrep_rule = self._parse_rule_metadata(raw_check, check_file, semgrep_rule)
+            check_definition = raw_check['definition']
+            if raw_check.get('mode', '') == 'taint':
+                semgrep_rule['mode'] = 'taint'
+                semgrep_rule.update(self._parse_taint_mode_definition(check_definition))
+            else:
+                semgrep_rule.update(self._parse_definition(check_definition))
+        except Exception as e:
+            logging.error(f'the policy in file {check_file} is misconfigured so it could not be parsed properly.\n{e}')
```

Comment:
would change it to just "the policy couldn't be parsed properly"
we might have valid policies which cannot be parsed to semgrep.